### ISSUE TITLE
debuginfo is not stripped when using `bazel build -c opt`

### ISSUE THREAD

**Issue by johnlanni** [proxy-wasm/proxy-wasm-rust-sdk#149]
build the hello_world example in the project:
```bash
bazel build -c opt //examples:hello_world
```

use `wasm2wat` to generate the wat file:
```bash
wasm2wat bazel-bin/examples/hello_world.wasm -o hello_world.wat
```

then you will find that the debuginfos is not stripped:
```text
...
  (import "wasi_snapshot_preview1" "environ_sizes_get" (func $__imported_wasi_snapshot_preview1_environ_sizes_get (type 4)))
  (import "wasi_snapshot_preview1" "proc_exit" (func $__imported_wasi_snapshot_preview1_proc_exit (type 2)))
  (func $__wasm_call_ctors (type 18)
    call $__wasilibc_initialize_environ_eagerly)
  (func $_ZN4core3ops8function6FnOnce9call_once17h2428394f5d7238e6E (type 3) (param i32 i32)
    local.get 0
    i32.const 1048980
    i32.store offset=4
    local.get 0
    i32.const 1
    i32.store)
  (func $_ZN4core3ptr44drop_in_place$LT$getrandom..error..Error$GT$17h161e8c8ed28efdccE (type 2) (param i32))
  (func $_ZN75_$LT$hello_world..HelloWorld$u20$as$u20$proxy_wasm..traits..RootContext$GT$11on_vm_start17h33e5123357b23fd4E (type 4) (param i32 i32) (result i32)
...
```

It increases the size of the wasm file and reduces performance. The `hello_world.wasm` is about 1.9Mb:
![image](https://user-images.githubusercontent.com/6763318/165010464-c79b8d80-699d-4da3-9023-9d019d2c73ab.png)

I use this project ([hello-world-wasm.zip](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/files/8551257/hello-world-wasm.zip)) and build the wasm file with `wasm-pack build --release`, the wasm file size is only 80Kb:

![image](https://user-images.githubusercontent.com/6763318/165011673-bf67e1c8-85d3-4568-b078-e808a8b245fe.png)

**Comment by johnlanni** [comment]
This problem has a big impact on performance. In my tests I found wasm plugins generated with `wasm-pack build --release` reduced latency by 300%

**Comment by PiotrSikora** [comment]
@johnlanni sorry for the delay.

Indeed, Bazel doesn't strip debug symbols, but that's really an issue with [`rules_rust`](https://github.com/bazelbuild/rules_rust) and not this SDK.

However, Cargo is the preferred method of building plugins using Proxy-Wasm Rust SDK, and Bazel integration exists only to make it easier to integrate with other Bazel projects (e.g. Envoy) when running tests, etc.

Anyway, I've ran some tests with Cargo to understand why `wasm-pack` produces so small files. You can replicate those results using commands below:
```sh
cargo build --quiet --release --example http_auth_random --target=wasm32-unknown-unknown
cargo build --quiet --release --example http_auth_random --target=wasm32-wasi
ls -1sh target/wasm32-*/release/examples/http_auth_random.wasm
```

`v0.2.0`:
```
276K target/wasm32-unknown-unknown/release/examples/http_auth_random.wasm
312K target/wasm32-wasi/release/examples/http_auth_random.wasm
```

With #160 (adding `codegen-units = 1` and `strip = "debuginfo"` added in Rust v1.59.0):
```
132K target/wasm32-unknown-unknown/release/examples/http_auth_random.wasm
156K target/wasm32-wasi/release/examples/http_auth_random.wasm
```

Addintionally replacing `strip = "debuginfo"` with `strip = "symbols"` (but this removes symbols and `name` section, which means that you won't get backtraces, and I doubt that it has any effect on the performance):
```
112K target/wasm32-unknown-unknown/release/examples/http_auth_random.wasm
132K target/wasm32-wasi/release/examples/http_auth_random.wasm
```

Additionally replacing `opt-level = 3` with `opt-level = "z"` (which I doubt results in better performance, but I'm happy to be proven otherwise and change it then):
```
 84K target/wasm32-unknown-unknown/release/examples/http_auth_random.wasm
108K target/wasm32-wasi/release/examples/http_auth_random.wasm
```

Basically, `wasm-pack` builds against `wasm32-unknown-unknown` target (losing access to environment variables, randomness, preopened files, and other WASI goodies in the future), strips all symbols (losing backtraces), and optimizes for size instead of speed.

I honestly don't think any of those tradeoffs (other than stripping the debug info) are worth it.

**Comment by PiotrSikora** [comment]
For the reference, debuginfo can be stripped from Bazel build with `rustc_flags = ["-Cstrip=debuginfo"]`.
